src/bucket/Bucket.cpp

```diff
@@ -82,19 +82,24 @@ Bucket::containsBucketIdentity(BucketEntry const& id) const
     return false;
 }
 
+#ifdef BUILD_TESTS
 void
 Bucket::apply(Application& app) const
 {
     ZoneScoped;
+
+    std::vector<uint8_t> filter(
```

Comment:
Why uint8_t instead of bool?

Reply:
I agree that predicates is much cleaner but probably not more efficient (I did the `vector` because of the very efficient look-up). But the reality is the interface is so much cleaner that it's worth it, and we could always wrap the vector approach in a function if profiling indicates that would be valuable. Done.